codalab/worker/docker_image_manager.py

```diff
@@ -243,10 +243,13 @@ def image_availability_state(image_spec, success_message, failure_message):
                 def download():
                     logger.debug('Downloading Docker image %s', image_spec)
                     try:
-                        self._docker.images.pull(image_spec)
+                        client = docker.APIClient(base_url='unix://var/run/docker.sock')
+                        self.generator = client.pull(image_spec, stream=True, decode=True)
```

Comment:
Why do we need to create a new client as opposed to just running `self._docker.images.pull`?

Reply:
When I do this, the `generator` returned is not an iterator. I am not sure why this is the case - but it only works when you have a new `APIClient`

Reply:
Ok. What exactly is returned from the generator coming from `self._docker.images.pull`? Is `self._docker.images` also an `APIClient`?

Reply:
its an `ImageCollection` I believe - I will have to double check.

Reply:
It looks like when we use `self._docker.images.pull`, it calls the same `pull` as we do with the new `APIClient`, but omits the output.

Reference: https://github.com/docker/docker-py/blob/master/docker/models/images.py#L449

Reply:
Ah, I see. In that case, can we just replace this with `client.client.api.pull` then? This way, we don't have to instantiate a new `APIClient`.